src/marqo/core/index_management/vespa_application_package.py

```diff
@@ -0,0 +1,644 @@
+
+            self._index_setting_store.save_index_setting(index)
+            self._store.save_file(schema, 'schemas', f'{index.schema_name}.sd')
+            self._service_xml.add_schema(index.schema_name)
+
+        self._persist_index_settings()
+        self._store.save_file(self._service_xml.to_xml(), self._SERVICES_XML_FILE)
+
+    def batch_delete_index_setting_and_schema(self, index_names: List[str]) -> None:
+        for name in index_names:
+            index = self._index_setting_store.get_index(name)
+            if index is None:
+                raise IndexNotFoundError(f"Index {name} not found")
+            self._index_setting_store.delete_index_setting(index.name)
+            self._store.remove_file('schemas', f'{index.schema_name}.sd')
+            self._service_xml.remove_schema(index.schema_name)
+
+        self._add_schema_removal_override()
+        self._persist_index_settings()
+        self._store.save_file(self._service_xml.to_xml(), self._SERVICES_XML_FILE)
+
+    def has_schema(self, name: str) -> bool:
+        return self._store.file_exists('schemas', f'{name}.sd')
+
+    def has_index(self, name: str) -> bool:
+        return self._index_setting_store.get_index(name) is not None
+
+    def _persist_index_settings(self):
+        index_setting_json, index_setting_history_json = self._index_setting_store.to_json()
+        self._store.save_file(index_setting_json, self._MARQO_INDEX_SETTINGS_FILE)
+        self._store.save_file(index_setting_history_json, self._MARQO_INDEX_SETTINGS_HISTORY_FILE)
+
+    def _config_query_profiles(self, backup: VespaAppBackup) -> None:
+        content = textwrap.dedent(
+            '''
+            <query-profile id="default">
+                <field name="maxHits">1000000</field>
+                <field name="maxOffset">1000000</field>
+            </query-profile>
+            '''
+        )
+        logger.info(f'Configuring query profiles {content}')
+        self._store.save_file(content, 'search', 'query-profiles', 'default.xml', backup=backup)
+
+    def _copy_components_jar(self) -> None:
+        components_jar_file = 'marqo-custom-searchers-deploy.jar'
+
+        logger.info(f'Copying components jar file {components_jar_file}')
+        with open(self._COMPONENTS_JAR_FOLDER/components_jar_file, 'rb') as f:
+            self._store.save_file(f.read(), 'components', components_jar_file)
+
+    def _add_schema_removal_override(self) -> None:
+        content = textwrap.dedent(
+            f'''
+            <validation-overrides>
+                 <allow until='{datetime.utcnow().strftime('%Y-%m-%d')}'>schema-removal</allow>
+            </validation-overrides>
+            '''
+        ).strip()
+        self._store.save_file(content, 'validation-overrides.xml')
+
+    def _validate_services_xml_for_rollback(self, services_xml_backup: str) -> None:
+        services_xml_old = ServiceXml(services_xml_backup)
```

Comment:
During the rollback, if we detect the changes in 'content/documents' or any 'nodes' elements in services.xml, we will abort the rollback. This is to avoid rolling back to an inconsistent state. Should we capture any diffs of the services.xml file in the log for debugging purpose? @farshidz